src/components/Misc/DeleteOrResetModal.tsx

```diff
@@ -0,0 +1,130 @@
+import { Box, Button, Modal, Stack, Text, TextInput } from '@mantine/core';
+import classes from './styles/DeleteOrResetModal.module.css';
+import { ChangeEvent, useCallback, useState } from 'react';
+
+type BaseProps = {
+	isOpen: boolean;
+	onClose: () => void;
+	header: string;
+	specialContent?: React.ReactNode;
+	content: string;
+	actionProcessingContent?: React.ReactNode;
+	isActionInProgress?: boolean;
+	onConfirm: () => void;
+};
+
+// Note: The `confirmationText` and `placeholder` props are required for 'delete' and 'reset' types, but not for 'simple' type.
+type DeleteOrResetModalProps =
+	| (BaseProps & {
+			type: 'simple';
+			confirmationText?: never; // Will throw an error if `confirmationText` is passed
+			placeholder?: never;
+	  })
+	| (BaseProps & {
+			type: 'delete' | 'reset';
+			confirmationText: string;
+			placeholder: string;
+	  });
+
+/**
+ * Confirmation modal for deleting or resetting an item.
+ * @param type - Specifies the type of modal ('simple', 'delete', or 'reset').
+ * @param isOpen - Controls whether the modal is visible.
+ * @param onClose - Callback to close the modal and reset the state.
+ * @param header - Header text displayed in the modal title.
+ * @param specialContent - Optional content for additional context or customization.
+ * @param content - Main descriptive content of the modal.
+ * @param placeholder - Input placeholder for confirmation text (applicable to 'delete' and 'reset').
+ * @param confirmationText - Text required to confirm the action (applicable to 'delete' and 'reset').
+ * @param actionProcessingContent - Optional content below text input for showing progress status or related information.
+ * @param isActionInProgress - Disables the confirm button when action is in progress.
+ * @param onConfirm - Callback function to be executed when the confirm button is clicked.
+ */
+const DeleteOrResetModal = ({
+	type,
+	isOpen,
+	onClose,
+	header,
+	specialContent,
+	content,
+	placeholder,
+	confirmationText,
+	actionProcessingContent,
+	isActionInProgress,
+	onConfirm,
+}: DeleteOrResetModalProps) => {
+	const [confirmText, setConfirmText] = useState<string>('');
+
+	// Handler for the confirmation input field
+	const onChangeHandler = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+		setConfirmText(e.target.value);
+	}, []);
+
+	// Function to validate and trigger confirmation logic
+	const tryConfirm = useCallback(() => {
+		if (type === 'simple' || confirmationText === confirmText) {
+			setConfirmText('');
+			onConfirm();
+		}
+	}, [type, confirmationText, confirmText, onConfirm]);
+
+	// Function to close the modal and reset the confirmation text state.
+	const closeModal = useCallback(() => {
+		setConfirmText('');
+		onClose();
+	}, [onClose]);
+
+	return (
+		<Modal
+			withinPortal
+			opened={isOpen}
+			onClose={closeModal}
+			size="auto"
+			centered
+			styles={{
+				body: { padding: '0 1rem 1rem 1rem', width: 400 },
```

Comment:
Move this to styles file using a className.

Reply:
I’ve moved the styles to the CSS module file. I used the mantine [`classNames` feature](https://mantine.dev/styles/styles-api/#classnames-prop) to do this.